x4.md

```diff
@@ -54,325 +49,93 @@ The protocol should be a reflection of our principles.
 
-Alternative method is to use [`shh_newMessageFilter`](https://github.com/ethereum/go-ethereum/wiki/Whisper-v6-RPC-API#shh_newmessagefilter) which takes the same criteria object and then periodically calling [`shh_getFilterMessages`](https://github.com/ethereum/go-ethereum/wiki/Whisper-v6-RPC-API#shh_getfiltermessages) method.
+| Layer             | Purpose                         | Technology                   |
+|-------------------|---------------------------------|------------------------------|
+| Data              | End user functionality          | 1:1, group chat, public chat |
+| Security          | Confidentiality, PFS            | Double Ratchet               |
+| Privacy           | Metadata protection             | Whisper                      |
+| P2P               | Overlay routing, NAT traversal  | devp2p                       |
 
-Learn more following [Whisper V6 RPC API](https://github.com/ethereum/go-ethereum/wiki/Whisper-v6-RPC-API).
+Note that the Status Protocol was documented after the fact, which might lead to some inconsistencies or questionable design choices.
 
-# Public messages
+# Data layer
 
-Public messages are encrypted with a symmetric key which is publicly known so anyone can participate in the conversation.
+Data layer specifies a format of payloads exchanged between clients communicating using the Status Protocol. The robust and well-defined format is required in order to provide some security guarantees, for example, client should not display messages that do not conform to the format, and allow the clients to upgrade the protocol safely in the future.
 
-The fact that anyone can participate makes the public chats voulnerable to spam attacks. Also, there are no moderators of these chats.
+The well-defined format also makes it possible to support multiple types of conversations as well as multiple types of content.
 
-## Sending
+Please refer to [Initial Message Payload Specification](x8.md) for more details.
 
-1. Calculate a symmetric key using [`shh_generateSymKeyFromPassword`](https://github.com/ethereum/go-ethereum/wiki/Whisper-v6-RPC-API#shh_generatesymkeyfrompassword) passing a public chat name as a string and save the result to `symKeyID`
-2. Call [`shh_post`](https://github.com/ethereum/go-ethereum/wiki/Whisper-v6-RPC-API#shh_post) with the following settings:
-   1. `symKeyID` MUST be set to `symKeyID`
-   2. `sig` MUST be set to `sigKeyID`
-   3. `ttl` MUST be at least `10` (it is in seconds)
-   4. `topic` MUST be set accordingly to [Topic](#topic) section and hex-encoded,
-   5. `payload` MUST be a hex-encoded string,
-   6. `powTime` MAY be arbitrary but should be enough to perform proof-of-work
-   7. `powTarget` MUST be equal or lower than `0.002`.
+# Security layer
 
-Learn more following [Whisper V6 RPC API](https://github.com/ethereum/go-ethereum/wiki/Whisper-v6-RPC-API).
+Security layer provides various cryptographical properties to the Status Protocol:
+* encryption
```

Comment:
Confidentiality
Integrity?

Reply:
Copy pasted from  http://cacr.uwaterloo.ca/techreports/2015/cacr2015-02.pdf - probably better to use accepted terms here

Confidentiality:Only  the  intended  recipients  are  able  toread a message. Specifically, the message must not be readableby a server operator that is not a conversation participant.Integrity:No  honest  party  will  accept  a  message  that  hasbeen modified in transit.Authentication:Each  participant  in  the  conversation  re-ceives proof of possession of a known long-term secret fromall  other  participants  that  they  believe  to  be  participating  inthe conversation. In addition, each participant is able to verifythat a message was sent from the claimed source.

Reply:
That's a good one!